### DISCUSSION TITLE
如何清空表单？

### DISCUSSION THREAD

**Discussion by yunuo-tech** [alibaba/formily#1569] [Q&A 2.x] [answered]
现在碰到一个情况：录入表单的时候 切换到其他页面后再切换回来，发现之前录入的信息还在，如何在离开当前页面后自动清空表单里的内容？ 用了form.setValues({})和form.setInitialValues({})都不行

**Comment by janryWang** [comment, marked as answer]
form.reset('*',{forceClear:true}) 或者form.setValues({},'overwrite') ，文档都写有，是文档写的有问题吗？

**Reply by yunuo-tech** [reply]
> form.reset('*',{forceClear:true}) 或者form.setValues({},'overwrite') ，文档都写有，是文档写的有问题吗？

事实上并不能清空，但是我目前不确定是哪个模块的bug，官网的例子在我本地也有问题：切换页面时，表单上的数据都无法清空，下次打开表单数据还是上一次录入的。

**Comment by yunuo-tech** [comment]
https://user-images.githubusercontent.com/5960864/121763092-e5aa1400-cb6b-11eb-9cf9-bbfc6217eb67.mp4

**Reply by janryWang** [reply]
你应该是跨页面共享createForm的实例了

**Reply by yunuo-tech** [reply]
> 你应该是跨页面共享createForm的实例了

复现例子：
https://codesandbox.io/s/nice-edison-kvsv8?file=/src/pages/index.tsx

**Comment by janryWang** [comment]
不要在全局createForm，正确的应该是在渲染函数内部使用useMemo创建form实例，它会随着页面切换动态回收
这样写
```
export default ()=>{
   const form = useMemo(()=>createForm(),[])
}
```

**Reply by yunuo-tech** [reply]
好的，谢谢。这一点我觉得有必要在文档里说明

**Reply by janryWang** [reply]
disucussion里有记录就行，文档没地方说明的
